### DISCUSSION TITLE
Understanding what this error means

### DISCUSSION THREAD

**Discussion by Livefour2day** [authelia/authelia#6865] [Q&A] [answered]
Just updated Authelia to V4.38.0 and get a fatal error just trying to understand what has gone wrong if anyone could help, this is from my logs -
time="2024-03-14T21:23:02Z" level=fatal msg="One or more providers had fatal failures performing startup checks, for more detail check the error level logs" providers="[storage]" stack="github.com/authelia/authelia/v4/internal/commands/root.go:149 doStartupChecks\ngithub.com/authelia/authelia/v4/internal/commands/root.go:87  (*CmdCtx).RootRunE\ngithub.com/spf13/cobra@v1.8.0/command.go:983                  (*Command).execute\ngithub.com/spf13/cobra@v1.8.0/command.go:1115                 (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.0/command.go:1039                 (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10       main\nruntime/internal/atomic/types.go:194                          (*Uint32).Load\nruntime/asm_amd64.s:1695                                      goexit"

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

The error message you're encountering indicates that there was a fatal failure with one or more providers during the startup checks of Authelia, specifically with the "storage" provider. This type of error typically occurs when there's an issue with the configuration or accessibility of the storage backend that Authelia is trying to use.

To troubleshoot this issue, you should:

1. **Check the Configuration**: Ensure that your configuration for the storage provider is correct. This includes checking any paths, credentials, or settings specific to the storage method you're using (e.g., SQLite, MySQL, PostgreSQL, etc.).

2. **Review Detailed Logs**: The message suggests checking the error level logs for more detail. Look for any error messages that precede this fatal error in the logs. These messages can provide more specific information about what went wrong with the storage provider.

3. **Verify Storage Accessibility**: Make sure that Authelia can access the storage backend. If you're using a database, ensure that the database is running and that network policies or firewalls are not blocking the connection.

4. **Consult Authelia Documentation**: Review the [Authelia documentation](https://www.authelia.com/docs/) for guidance on configuring the storage provider. There might be specific requirements or steps that you missed.

5. **Check for Known Issues**: Look at the [Authelia GitHub issues](https://github.com/authelia/authelia/issues) to see if others have encountered similar problems and if there are any suggested solutions or workarounds.

Given the timing of your error and the recent update to Authelia version 4.38.0, it's also possible that there are changes or deprecations in this version that affect the storage provider configuration. Review the [release notes for version 4.38.0](https://www.authelia.com/blog/4.38-release-notes) to see if there are any relevant notes regarding storage providers.

If after these steps you're still unable to resolve the issue, consider reaching out to the Authelia community for support, either through their [GitHub discussions](https://github.com/authelia/authelia/discussions) or other community channels.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by james-d-elliott** [comment]
Show the complete logs.

**Reply by Livefour2day** [reply]
text  error  warn  system  array  login  

time="2024-03-14T21:19:25Z" level=debug msg="Process user information" gid=99 uid=99
time="2024-03-14T21:19:25Z" level=warning msg="Configuration: configuration key 'storage.mysql.host' is deprecated in 4.38.0 and has been replaced by 'storage.mysql.address' when combined with the 'storage.mysql.port' in the format of '[tcp://]<hostname>[:<port>]': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:19:25Z" level=warning msg="Configuration: configuration key 'server.port' is deprecated in 4.38.0 and has been replaced by 'server.address' when combined with the 'server.host' and 'server.path' in the format of '[tcp[(4|6)]://]<hostname>[:<port>][/<path>]' or 'tcp[(4|6)://][hostname]:<port>[/<path>]': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:19:25Z" level=warning msg="Configuration: configuration key 'notifier.smtp.host' is deprecated in 4.38.0 and has been replaced by 'notifier.smtp.address' when combined with the 'notifier.smtp.port' in the format of '[tcp://]<hostname>[:<port>]': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:19:25Z" level=warning msg="Configuration: configuration key 'server.enable_expvars' is deprecated in 4.38.0 and has been replaced by 'server.endpoints.enable_expvars': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:19:25Z" level=warning msg="Configuration: configuration key 'authentication_backend.ldap.username_attribute' is deprecated in 4.38.0 and has been replaced by 'authentication_backend.ldap.attributes.username': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:19:25Z" level=warning msg="Configuration: configuration key 'server.enable_pprof' is deprecated in 4.38.0 and has been replaced by 'server.endpoints.enable_pprof': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:19:25Z" level=warning msg="Configuration: configuration key 'notifier.smtp.port' is deprecated in 4.38.0 and has been replaced by 'notifier.smtp.address' when combined with the 'notifier.smtp.host' in the format of '[tcp://]<hostname>[:<port>]': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:19:25Z" level=warning msg="Configuration: configuration key 'server.path' is deprecated in 4.38.0 and has been replaced by 'server.address' when combined with the 'server.host' and 'server.port' in the format of '[tcp[(4|6)]://]<hostname>[:<port>][/<path>]' or 'tcp[(4|6)://][hostname]:<port>[/<path>]': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:19:25Z" level=warning msg="Configuration: configuration key 'storage.mysql.port' is deprecated in 4.38.0 and has been replaced by 'storage.mysql.address' when combined with the 'storage.mysql.host' in the format of '[tcp://]<hostname>[:<port>]': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:19:25Z" level=warning msg="Configuration: configuration key 'session.remember_me_duration' is deprecated in 4.38.0 and has been replaced by 'session.remember_me': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:19:25Z" level=warning msg="Configuration: configuration key 'authentication_backend.ldap.group_name_attribute' is deprecated in 4.38.0 and has been replaced by 'authentication_backend.ldap.attributes.group_name': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:19:25Z" level=warning msg="Configuration: configuration key 'jwt_secret' is deprecated in 4.38.0 and has been replaced by 'identity_validation.reset_password.jwt_secret': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:19:25Z" level=warning msg="Configuration: configuration key 'authentication_backend.ldap.mail_attribute' is deprecated in 4.38.0 and has been replaced by 'authentication_backend.ldap.attributes.mail': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:19:25Z" level=warning msg="Configuration: configuration key 'authentication_backend.ldap.url' is deprecated in 4.38.0 and has been replaced by 'authentication_backend.ldap.address': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:19:25Z" level=warning msg="Configuration: configuration key 'server.host' is deprecated in 4.38.0 and has been replaced by 'server.address' when combined with the 'server.port' and 'server.path' in the format of '[tcp[(4|6)]://]<hostname>[:<port>][/<path>]' or 'tcp[(4|6)://][hostname]:<port>[/<path>]': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:19:25Z" level=warning msg="Configuration: configuration key 'authentication_backend.ldap.display_name_attribute' is deprecated in 4.38.0 and has been replaced by 'authentication_backend.ldap.attributes.display_name': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:19:25Z" level=warning msg="Configuration: session: option 'domain' is deprecated in v4.38.0 and has been replaced by a multi-domain configuration: this has automatically been mapped for you but you will need to adjust your configuration to remove this message and receive the latest messages"
time="2024-03-14T21:19:25Z" level=info msg="Authelia v4.38.0 is starting"
time="2024-03-14T21:19:25Z" level=info msg="Log severity set to debug"
time="2024-03-14T21:19:25Z" level=info msg="Storage schema is being checked for updates"
time="2024-03-14T21:19:25Z" level=info msg="Storage schema migration from 11 to 15 is being attempted"
time="2024-03-14T21:19:25Z" level=error msg="Error occurred running a startup check" error="error during schema migrate: error applying migration version 12 to version 11 for rollback: schema migration 12 (WebAuthnMultiCookieDomain) failed: Error 1061 (42000): Duplicate key name 'webauthn_devices_kid_key'. rollback caused by: schema migration 12 (WebAuthnMultiCookieDomain) failed: Error 1061 (42000): Duplicate key name 'webauthn_credentials_kid_key'" provider=storage stack="github.com/authelia/authelia/v4/internal/commands/root.go:107 doStartupChecks\ngithub.com/authelia/authelia/v4/internal/commands/root.go:87  (*CmdCtx).RootRunE\ngithub.com/spf13/cobra@v1.8.0/command.go:983                  (*Command).execute\ngithub.com/spf13/cobra@v1.8.0/command.go:1115                 (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.0/command.go:1039                 (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10       main\nruntime/internal/atomic/types.go:194                          (*Uint32).Load\nruntime/asm_amd64.s:1695                                      goexit"
time="2024-03-14T21:19:25Z" level=debug msg="LDAP Supported OIDs. Control Types: none. Extensions: 1.3.6.1.4.1.4203.1.11.1"
time="2024-03-14T21:19:25Z" level=debug msg="notification provider: startup check skipped as it is disabled"
time="2024-03-14T21:19:25Z" level=fatal msg="One or more providers had fatal failures performing startup checks, for more detail check the error level logs" providers="[storage]" stack="github.com/authelia/authelia/v4/internal/commands/root.go:149 doStartupChecks\ngithub.com/authelia/authelia/v4/internal/commands/root.go:87  (*CmdCtx).RootRunE\ngithub.com/spf13/cobra@v1.8.0/command.go:983                  (*Command).execute\ngithub.com/spf13/cobra@v1.8.0/command.go:1115                 (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.0/command.go:1039                 (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10       main\nruntime/internal/atomic/types.go:194                          (*Uint32).Load\nruntime/asm_amd64.s:1695                                      goexit"
time="2024-03-14T21:23:02Z" level=debug msg="Process user information" gid=99 uid=99
time="2024-03-14T21:23:02Z" level=warning msg="Configuration: configuration key 'server.host' is deprecated in 4.38.0 and has been replaced by 'server.address' when combined with the 'server.port' and 'server.path' in the format of '[tcp[(4|6)]://]<hostname>[:<port>][/<path>]' or 'tcp[(4|6)://][hostname]:<port>[/<path>]': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:23:02Z" level=warning msg="Configuration: configuration key 'authentication_backend.ldap.group_name_attribute' is deprecated in 4.38.0 and has been replaced by 'authentication_backend.ldap.attributes.group_name': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:23:02Z" level=warning msg="Configuration: configuration key 'server.port' is deprecated in 4.38.0 and has been replaced by 'server.address' when combined with the 'server.host' and 'server.path' in the format of '[tcp[(4|6)]://]<hostname>[:<port>][/<path>]' or 'tcp[(4|6)://][hostname]:<port>[/<path>]': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:23:02Z" level=warning msg="Configuration: configuration key 'authentication_backend.ldap.display_name_attribute' is deprecated in 4.38.0 and has been replaced by 'authentication_backend.ldap.attributes.display_name': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:23:02Z" level=warning msg="Configuration: configuration key 'storage.mysql.port' is deprecated in 4.38.0 and has been replaced by 'storage.mysql.address' when combined with the 'storage.mysql.host' in the format of '[tcp://]<hostname>[:<port>]': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:23:02Z" level=warning msg="Configuration: configuration key 'authentication_backend.ldap.url' is deprecated in 4.38.0 and has been replaced by 'authentication_backend.ldap.address': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:23:02Z" level=warning msg="Configuration: configuration key 'session.remember_me_duration' is deprecated in 4.38.0 and has been replaced by 'session.remember_me': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:23:02Z" level=warning msg="Configuration: configuration key 'server.enable_expvars' is deprecated in 4.38.0 and has been replaced by 'server.endpoints.enable_expvars': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:23:02Z" level=warning msg="Configuration: configuration key 'jwt_secret' is deprecated in 4.38.0 and has been replaced by 'identity_validation.reset_password.jwt_secret': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:23:02Z" level=warning msg="Configuration: configuration key 'authentication_backend.ldap.username_attribute' is deprecated in 4.38.0 and has been replaced by 'authentication_backend.ldap.attributes.username': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:23:02Z" level=warning msg="Configuration: configuration key 'server.enable_pprof' is deprecated in 4.38.0 and has been replaced by 'server.endpoints.enable_pprof': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:23:02Z" level=warning msg="Configuration: configuration key 'authentication_backend.ldap.mail_attribute' is deprecated in 4.38.0 and has been replaced by 'authentication_backend.ldap.attributes.mail': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:23:02Z" level=warning msg="Configuration: configuration key 'notifier.smtp.host' is deprecated in 4.38.0 and has been replaced by 'notifier.smtp.address' when combined with the 'notifier.smtp.port' in the format of '[tcp://]<hostname>[:<port>]': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:23:02Z" level=warning msg="Configuration: configuration key 'storage.mysql.host' is deprecated in 4.38.0 and has been replaced by 'storage.mysql.address' when combined with the 'storage.mysql.port' in the format of '[tcp://]<hostname>[:<port>]': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:23:02Z" level=warning msg="Configuration: configuration key 'server.path' is deprecated in 4.38.0 and has been replaced by 'server.address' when combined with the 'server.host' and 'server.port' in the format of '[tcp[(4|6)]://]<hostname>[:<port>][/<path>]' or 'tcp[(4|6)://][hostname]:<port>[/<path>]': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:23:02Z" level=warning msg="Configuration: configuration key 'notifier.smtp.port' is deprecated in 4.38.0 and has been replaced by 'notifier.smtp.address' when combined with the 'notifier.smtp.host' in the format of '[tcp://]<hostname>[:<port>]': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-03-14T21:23:02Z" level=warning msg="Configuration: session: option 'domain' is deprecated in v4.38.0 and has been replaced by a multi-domain configuration: this has automatically been mapped for you but you will need to adjust your configuration to remove this message and receive the latest messages"
time="2024-03-14T21:23:02Z" level=info msg="Authelia v4.38.0 is starting"
time="2024-03-14T21:23:02Z" level=info msg="Log severity set to debug"
time="2024-03-14T21:23:02Z" level=info msg="Storage schema is being checked for updates"
time="2024-03-14T21:23:02Z" level=info msg="Storage schema migration from 11 to 15 is being attempted"
time="2024-03-14T21:23:02Z" level=error msg="Error occurred running a startup check" error="error during schema migrate: error applying migration version 12 to version 11 for rollback: schema migration 12 (WebAuthnMultiCookieDomain) failed: Error 1061 (42000): Duplicate key name 'webauthn_devices_kid_key'. rollback caused by: schema migration 12 (WebAuthnMultiCookieDomain) failed: Error 1061 (42000): Duplicate key name 'webauthn_credentials_kid_key'" provider=storage stack="github.com/authelia/authelia/v4/internal/commands/root.go:107 doStartupChecks\ngithub.com/authelia/authelia/v4/internal/commands/root.go:87  (*CmdCtx).RootRunE\ngithub.com/spf13/cobra@v1.8.0/command.go:983                  (*Command).execute\ngithub.com/spf13/cobra@v1.8.0/command.go:1115                 (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.0/command.go:1039                 (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10       main\nruntime/internal/atomic/types.go:194                          (*Uint32).Load\nruntime/asm_amd64.s:1695                                      goexit"
time="2024-03-14T21:23:02Z" level=debug msg="LDAP Supported OIDs. Control Types: none. Extensions: 1.3.6.1.4.1.4203.1.11.1"
time="2024-03-14T21:23:02Z" level=debug msg="notification provider: startup check skipped as it is disabled"
time="2024-03-14T21:23:02Z" level=fatal msg="One or more providers had fatal failures performing startup checks, for more detail check the error level logs" providers="[storage]" stack="github.com/authelia/authelia/v4/internal/commands/root.go:149 doStartupChecks\ngithub.com/authelia/authelia/v4/internal/commands/root.go:87  (*CmdCtx).RootRunE\ngithub.com/spf13/cobra@v1.8.0/command.go:983                  (*Command).execute\ngithub.com/spf13/cobra@v1.8.0/command.go:1115                 (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.0/command.go:1039                 (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10       main\nruntime/internal/atomic/types.go:194                          (*Uint32).Load\nruntime/asm_amd64.s:1695                                      goexit"

** Press ANY KEY to close this window **

**Comment by james-d-elliott** [comment, marked as answer]
It means you're also experiencing this issue and we need the original log messages: https://github.com/authelia/authelia/issues/6859
